src/Solvers/matrix_solver_utils.jl

```diff
@@ -154,6 +154,40 @@ function compute_matrix_for_linear_operation(::CPU, template_field, linear_opera
     return A
 end
 
+function compute_matrix_for_linear_operation_kernel!(Aeᵢⱼₖ, eᵢⱼₖ, linear_operation!, colptr, rowval, nzval, args...)
+    i, j, k = (blockIdx().x - 1) * blockDim().x + threadIdx().x,
+              (blockIdx().y - 1) * blockDim().y + threadIdx().y,
+              (blockIdx().z - 1) * blockDim().z + threadIdx().z
```

Comment:
is it possible to use `KernelAbstractions`?

Reply:
I have updated the code to use `KernelAbstractions`.